### PR TITLE
new: added support for armv7-unknown-linux-gnueabi and armv7-unknown-linux-gnueabihf (closes #22)

### LINKED ISSUE #22: add support for armv7
Hi again, currently `aya::generated` only supports `x86_64` and `aarch64`.

If I try to compile with `cargo build --target=armv7-unknown-linux-gnueabihf` for instance (Raspberry Pi 4 target), the build fails with:

```
error[E0432]: unresolved imports `crate::generated::bpf_map_type`, `crate::generated::AYA_PERF_EVENT_IOC_DISABLE`, `crate::generated::AYA_PERF_EVENT_IOC_ENABLE`, `crate::generated::AYA_PERF_EVENT_IOC_SET_BPF`
  --> /home/evilsocket/.cargo/git/checkouts/aya-92d3024d6eb5b473/c240a2c/aya/src/bpf.rs:13:9
   |
13 |         bpf_map_type::BPF_MAP_TYPE_PERF_EVENT_ARRAY, AYA_PERF_EVENT_IOC_DISABLE,
   |         ^^^^^^^^^^^^                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^ no `AYA_PERF_EVENT_IOC_DISABLE` in `generated`
   |         |
   |         could not find `bpf_map_type` in `generated`
14 |         AYA_PERF_EVENT_IOC_ENABLE, AYA_PERF_EVENT_IOC_SET_BPF,
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^^^^^^^^^ no `AYA_PERF_EVENT_IOC_SET_BPF` in `generated`
   |         |
   |         no `AYA_PERF_EVENT_IOC_ENABLE` in `generated`

error[E0432]: unresolved import `crate::generated::bpf_map_type`
  --> /home/evilsocket/.cargo/git/checkouts/aya-92d3024d6eb5b473/c240a2c/aya/src/maps/mod.rs:42:5
   |
42 |     generated::bpf_map_type,
   |     ^^^^^^^^^^^^^^^^^^^^^^^ no `bpf_map_type` in `generated`

....
....
a lot more
....
....
```

Any chance you can either add support for armv7, or alternatively guide me to generate those files and send a PR? 

If it is as simple as just running `rust-bindgen` (on what?), maybe it's also worth adding support for other tier 1 and tier 2 targets?

Thanks.

### PR DESCRIPTION
**Pull request by evilsocket**

### CODE REVIEW

File: aya/src/generated/mod.rs [hunk 1]
```diff
@@ -3,6 +3,8 @@
 mod btf_internal_bindings;
 #[cfg(target_arch = "aarch64")]
 mod linux_bindings_aarch64;
+#[cfg(target_arch = "arm")]
```
Comment (alessandrod) [inline aya/src/generated/mod.rs:6]: shouldn't this be armv7?

File: aya/src/maps/perf/perf_buffer.rs [hunk 2]
```diff
@@ -273,14 +273,19 @@ impl Drop for PerfBuffer {
     }
 }
 
+#[cfg(target_pointer_width = "32")]
+type OffsetType = i32;
+#[cfg(target_pointer_width = "64")]
+type OffsetType = i64;
```
Comment (alessandrod) [inline aya/src/maps/perf/perf_buffer.rs:279]: I think that this should be `libc::off_t`?

File: aya/src/maps/mod.rs [hunk 3]
```diff
@@ -84,7 +90,7 @@ pub enum MapError {
     #[error("failed to create map `{name}`: {code}")]
     CreateError {
         name: String,
-        code: i64,
+        code: CodeType,
```
Comment (alessandrod) [inline aya/src/maps/mod.rs:93]: I think this should be `c_long`?

File: aya/src/maps/mod.rs [hunk 4]
```diff
@@ -141,7 +147,7 @@ impl Map {
 
         let fd = bpf_create_map(&c_name, &self.obj.def).map_err(|(code, io_error)| {
             MapError::CreateError {
-                name,
+                name: name,
```
Comment (alessandrod) [inline aya/src/maps/mod.rs:150]: undo this please

File: aya/src/sys/mod.rs [hunk 5]
```diff
@@ -64,7 +69,7 @@ unsafe fn syscall_impl(call: Syscall) -> SysResult {
             flags,
         } => libc::syscall(SYS_perf_event_open, &attr, pid, cpu, group, flags),
         PerfEventIoctl { fd, request, arg } => {
-            libc::ioctl(fd, request.try_into().unwrap(), arg) as i64
+            libc::ioctl(fd, request.try_into().unwrap(), arg) as RetType
```
Comment (alessandrod) [inline aya/src/sys/mod.rs:72]: I think this should be `c_long`

**Review by alessandrod — CHANGES_REQUESTED** [review]

File: aya/src/maps/perf/perf_buffer.rs (hunk 2 above)
Comment (evilsocket) [reply aya/src/maps/perf/perf_buffer.rs:279]: it was declared as i64 and i've put the conditional definition, i can change it to libc::off_t

File: aya/src/sys/mod.rs (hunk 5 above)
Comment (evilsocket) [reply aya/src/sys/mod.rs:72]: Same as before, it was defined as i64, I'll use c_long

File: aya/src/maps/perf/perf_buffer.rs (hunk 2 above)
Comment (alessandrod) [reply aya/src/maps/perf/perf_buffer.rs:279]: Yes please - here and in the other cases you found I was sloppy and used rust primitive types where I should have used libc types

Comment (evilsocket) [reply aya/src/maps/perf/perf_buffer.rs:279]: fair enough, will do asap :)

File: aya/src/generated/mod.rs (hunk 1 above)
Comment (evilsocket) [reply aya/src/generated/mod.rs:6]: nope, for rust target_arm==armv7 https://doc.rust-lang.org/reference/conditional-compilation.html#target_arch

**Comment by evilsocket** [comment]
added the cargo config for armv7-unknown-linux-gnueabihf

**Comment by evilsocket** [comment]
@alessandrod this still doesn't build on armv7-unknown-linux-gnueabihf native, wait to merge

**Comment by alessandrod** [comment]
Amazing work, thanks you 🙏

**Comment by evilsocket** [comment]
My pleasure!

**Pull request merged**